### DISCUSSION TITLE
Is there a way to have multiple view options when executing?

### DISCUSSION THREAD

**Discussion by dachmu** [michaelb/sniprun#181] [Q&A] [answered]
I do a lot of scripting, and this includes running some commands a lot of times, some of them have very short outputs, which the default view is lovely, but, some of them, like ansible commands over hundreds of servers, have huge outputs, and I would like to choose between a normal execute or a different type, like sending it to split terminal, or a tmux pane/window, and or possible saving the execution output to a file or buffer, I would love to be able to have different keymaps for the above behavior, and have not be able to find how to do that. Thanks :)

**Comment by michaelb** [comment, marked as answer]
Unfortunately, every existing solution has a caveat.

The best I can propose without having to code something from scratch would be this:

it's possible to toggle an alternate configurations on and off, like so:

`:lua require('sniprun').setup({ option={"i want", "to change"} })`

`:lua require('sniprun').setup({ option={"to the old one"} })`

and bind that to commands (on/off, or even better , an unique command with a pseudo-global switch; A slightly more compact version would have to be a full-fledged function that you would have to put into your (lua) configuration and bind a command/keymap on it.

``` lua

function toggle()
    local s = require'sniprun'.config_values
    if s.switch_configs == nil then -- the name 'switch_configs' is arbitrary, but choose something that 
                                                     -- doesn't conflict with a config option
        s.switch_configs = "not nil"
        s.display = { "Terminal", "VirtualTextErr" }
        print("Switched to alternate config")
    else
        s.display = { "Classic", "VirtualTextOk" }
        s.switch_configs = nil
        print("Switched to main config")
    end
end
```

The other ways of doing that would be to include a `s.run()` inside a command/script that would:

1. change the config to alternate values
2. lua require('sniprun').run()
3. change the config back

or 

1. get the current line number (or range) and filetype
2. Use the API to override the config `lua require('sniprun.api').run_range(line_nb, line_nb, filetype, {display={"Terminal"}})`

I'm currently revamping the display mode system, and I'm planning to add two new display modes (virtual line and file), in addition to being able to filter on 'ok' or 'error'. Though, for now, it's not possible to get the output in a file or a console pane 'natively' (well, there is an API, so everything is possible)

**Reply by dachmu** [reply]
Thank you, I appreciate the help, this is a great plugin, and I look forward for any future developments, it would be nice if there was a bit more in the way of documentation, like a wiki, for the API, I haven't been able to get my head around it and it sounds like it will allow a bit more flexibility in use. Thanks again and good day :)

**Reply by michaelb** [reply]
Don't hesitate to open a feature request or ask for a wiki in an issue. (especially during hacktoberfest. I might be a tad more motivated or someone may make a PR)

 They can help me know if many people 'like' the idea and you'll get notified if/when work is done
